cmd/template-resolver/utils/resolver_utils.go

```diff
@@ -137,26 +145,79 @@ func ProcessTemplate(yamlBytes []byte, hubKubeConfigPath, clusterName string) ([
 			LookupNamespace: policy.GetNamespace(),
 		}
 
-		hubResolver, err = templates.NewResolver(hubKubeConfig, hubTemplatesConfig)
+		if hubNS != "" {
+			hubTemplateOpts.opts.LookupNamespace = hubNS
+		}
```

Comment:
Oh I see what's going on now. I think it might be clearer to set `hubNS` in the conditionals above and set `LookupNamespace: hubNS` directly in the struct? I'll have to think through whether that matches the logic here...